src/loaders/dbm_load.c

```diff
@@ -321,22 +321,32 @@ static int get_smpl(struct module_data *m, int size, HIO_HANDLE *f, void *parm)
 static int get_venv(struct module_data *m, int size, HIO_HANDLE *f, void *parm)
 {
 	struct xmp_module *mod = &m->mod;
-	int i, j, nenv, ins;
+	int i, j, nenv, ins, flg, npt, sus, lps, lpe;
 
 	nenv = hio_read16b(f);
 
 	D_(D_INFO "Vol envelopes  : %d ", nenv);
 
 	for (i = 0; i < nenv; i++) {
 		ins = hio_read16b(f) - 1;
-		mod->xxi[ins].aei.flg = hio_read8(f) & 0x07;
-		mod->xxi[ins].aei.npt = hio_read8(f);
-		mod->xxi[ins].aei.sus = hio_read8(f);
-		mod->xxi[ins].aei.lps = hio_read8(f);
-		mod->xxi[ins].aei.lpe = hio_read8(f);
+		flg = hio_read8(f) & 0x07;
+		npt = hio_read8(f) + 1;
```

Comment:
should probably do the addition later to avoid integer overflow. nothing bad happens in this specific case but would still be cleaner.

Reply:
This is fixed now, thanks. I also applied this change to `ins`, where something bad *was* happening: it could resolve to -1 and crash due to another missing out-of-bounds write check.